src/absil/illib.fs

```diff
@@ -503,9 +503,20 @@ module String =
     let uppercase (s: string) =
         s.ToUpperInvariant()
 
-    let isUpper (s: string) = 
-        s.Length >= 1 && Char.IsUpper s.[0] && not (Char.IsLower s.[0])
-        
+    // Scripts that distinguish between upper and lower case (bicameral) DU Discriminators and Active Pattern identifiers are required to start with an upper case character.
+    // For valid identifiers where the case of the identifier can not be determined because there is no upper and lower case we will allow DU Discriminators and upper case characters 
+    // to be used.  This means that developers using unicameral scripts such as hindi, are not required to prefix these identifiers with an Upper case latin character. 
+    // '_' is a special case'  It is a valid identifier char, it disables definite assignment checks.  And it is neither upper or lower cased.
+    let isUpperCaseCharacter c =
+        // if IsUpper and IsLower return the same value, then we can't tell if it's upper or lower case, so it is a case insensensitive language
+        // thusly the char must be good.
+        if Char.IsUpper c = Char.IsLower c then (c <> '_')
```

Comment:
Could it be something like
```fsharp
if Char.IsUpper c = Char.IsLower c then Char.IsLetter c
```
instead? Allowing arbitrary symbols like `+` seems to be a too big breaking change to me.

Reply:
@auduchinok, I quite like that, however, Identifiers must begin with letters already and so it's only quoted identifiers that would really be impacted and we still need to cope with '_'.
But yes I will make that adjustment we have the restriction today, and keeping it as is, has a lot of value.